packages/axios-extension/src/auth/uaa.ts

```diff
@@ -105,6 +123,28 @@ export class Uaa {
         );
     }
 
+    /**
+     * Generate an access token using grant_type password to the authorization service (XSUAA).
+     *
+     * @returns an axios request config
+     */
+    protected getTokenRequestForClientCredential(): AxiosRequestConfig {
+        return {
+            url: this.url + '/oauth/token',
+            method: 'POST',
+            data: qs.stringify({
+                grant_type: 'password',
+                username: this.username,
+                password: this.password
+            }),
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+                Accept: 'application/json',
+                Authorization: `Basic ${Buffer.from(this.clientid + ':' + this.clientsecret).toString('base64')}`
```

Comment:
Maybe using a separate statement to avoid string concatenation.

```
const secret = `${this.clientid}:${this.clientsecret}`
....
Authorization: `Basic ${Buffer.from(secret).toString('base64')}`
```